### ISSUE TITLE
iOS failed to Compile after  1.3.0 

### ISSUE THREAD

**Issue by atrope** [RevenueCat/purchases-flutter#99]
Hello,
After updating to 1.3.0 iOS stopped building for with the following error:

```
Ld /Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Intermediates.noindex/Pods.build/Release-iphoneos/purchases_flutter.build/Objects-normal/arm64/Binary/purchases_flutter normal arm64 (in target 'purchases_flutter' from project 'Pods')
    cd /Users/trope/suamusica/OneApp/ios/Pods
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target arm64-apple-ios9.0 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.0.sdk -L/Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Products/Release-iphoneos/purchases_flutter -F/Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Products/Release-iphoneos/purchases_flutter -F/Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Products/Release-iphoneos/Purchases -F/Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Products/Release-iphoneos/PurchasesCoreSwift -F/Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Products/Release-iphoneos/PurchasesHybridCommon -F/Users/trope/suamusica/OneApp/ios/Pods/../Flutter -filelist /Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Intermediates.noindex/Pods.build/Release-iphoneos/purchases_flutter.build/Objects-normal/arm64/purchases_flutter.LinkFileList -install_name @rpath/purchases_flutter.framework/purchases_flutter -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Intermediates.noindex/Pods.build/Release-iphoneos/purchases_flutter.build/Objects-normal/arm64/purchases_flutter_lto.o -fobjc-arc -fobjc-link-runtime -framework Flutter -framework Foundation -framework PurchasesHybridCommon -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Intermediates.noindex/Pods.build/Release-iphoneos/purchases_flutter.build/Objects-normal/arm64/purchases_flutter_dependency_info.dat -o /Users/trope/Library/Developer/Xcode/DerivedData/Runner-ewvwfgyxollwiacgjqgpvwzsvjkh/Build/Intermediates.noindex/Pods.build/Release-iphoneos/purchases_flutter.build/Objects-normal/arm64/Binary/purchases_flutter

Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_RCPurchases", referenced from:
      objc-class-ref in PurchasesFlutterPlugin.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

I Had to downgrade in order to compile

How to fix that?

**Comment by aboedo** [comment]
this looks very similar to https://github.com/RevenueCat/purchases-flutter/issues/98, could you try using the steps described in that ticket?

**Comment by atrope** [comment]
Yes, i will try everything but minimum version to ios12 as we have users with ios9 :)

**Comment by eronnascimento** [comment]
Same error here :/ it is happening since my mac has updated xcode to version 12

**Comment by killmegenius** [comment]
Same error with physical iPad 3 iOS 9.3.5

**Comment by atrope** [comment]
@aboedo did not work.
When i downgrade to 1.2.1 everything build as expected.

**Comment by fuatkestane** [comment]
I'm taking same error. Sometimes it works, sometimes it doesn't. I couldn't make sense.

**Comment by eronnascimento** [comment]
> @aboedo did not work.
> When i downgrade to 1.2.1 everything build as expected.

Even drowngrading, I still get this error:
Undefined symbols for architecture arm64:
      "_OBJC_CLASS_$_RCPurchases", referenced from:
          objc-class-ref in PurchasesFlutterPlugin.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

**Comment by aboedo** [comment]
I'm currently investigating. Does this still happen on Xcode 11?

**Comment by aboedo** [comment]
I'm still trying to figure this out. I haven't been getting the same exact error as you, I got one that's almost the same except that it's when linking a different symbol, `OBJC_CLASS$_RCTransaction`. 
Could you try updating your Podfile and commenting out the line that reads `use_frameworks!` and see if it compiles after that? it seems to solve the problem on a couple of projects I'm testing with.

**Comment by atrope** [comment]
hi @aboedo,
I Need that line for other frameworks :/

**Comment by aboedo** [comment]
@atrope I figured that might be the case. I'm working on a fix, I'll get it in soon.

**Comment by eronnascimento** [comment]
I managed to compile without any problems using Xcode 11.7 and now I am changing my Xcode command line tools to this version so that I can use flutter build ios --release

**Comment by aboedo** [comment]
release 1.3.1 is out and should hopefully fix these issues, check it out and let me know how it goes!

**Comment by jacksonw765** [comment]
I am having this issue as well. No luck with any of the fixes mentioned.

**Comment by eronnascimento** [comment]
@jacksonw765 I solved this downgrading Xcode command line tools to 11.7, purchases_flutter to 1.2.1 and after that, do "flutter clean" and then "flutter run"

**Comment by eronnascimento** [comment]
@aboedo, I haven't tested yet and I ended up downgrading the version of xcode and purchases_flutter but as soon as possible I will upgrade the version and I give you some feedback.

**Comment by jacksonw765** [comment]
> @jacksonw765 I solved this downgrading Xcode command line tools to 11.7, purchases_flutter to 1.2.1 and after that, do "flutter clean" and then "flutter run"

I am an iOS novice so help me out here. (Android developer lol) I am on Xcode 12 am I able to downgrade the tools separately? Will I still be able to target iOS 14? Thanks for the help.

**Comment by jacksonw765** [comment]
I was able to fix it by pegging the version to master.
```
  purchases_flutter:
    git:
      url: https://github.com/RevenueCat/purchases-flutter.git
      ref: 3ff5c45
```

**Comment by aboedo** [comment]
@jacksonw765 if you downloaded Xcode through the App Store and you've already upgraded to Xcode 12, you can still use Xcode 11 by downloading it from Apple's downloads page here: https://developer.apple.com/download/more/. 

Builds made with Xcode 11 will still be accepted for a while, and iOS 14 users will be able to download apps made with Xcode 11. 
However, you won't be able to have code specifically built for iOS 14 (like iOS 14-exclusive APIs) without a bit more work, like for example a compiler flag. If you're not using something that's exclusive to iOS 14, you should be fine.

**Comment by atrope** [comment]
Hi, I was able to have a proper build :)

Thanks for the fix!

**Comment by aboedo** [comment]
@atrope awesome!! glad to hear it's working. I'm closing this for now, but if others are still struggling they should feel free to reopen. 

Thanks again to everyone for reporting! 🎉

**Comment by jacksonw765** [comment]
I know this is closed however I am still seeing the issue on Xcode 12.

**Comment by aboedo** [comment]
@jacksonw765 have you tried with 1.3.1?

**Comment by jacksonw765** [comment]
@aboedo yes, same issue.

**Comment by atrope** [comment]
@jacksonw765 try to comment then flutter pub get, then uncomment and another flutter pub get.

then in your ios folder, try to 

`
pod install --repo-update`

sometimes is cache.

**Comment by aboedo** [comment]
@jacksonw765 
I just noticed that you mentioned earlier that you fixed it by using `master`. `master` is deprecated and is way behind, which is why that was solved - we're using `main` as our stable branch. So I expect that you'd see the same issues if you use the `main` version. 

 try out @atrope's solution and we'll take it from there

**Comment by jacksonw765** [comment]
I'll give that a try! Thanks.

**Comment by jacksonw765** [comment]
No luck, same error. 

`ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
`

**Comment by aboedo** [comment]
is the error still pointing to the same symbol as before?

**Comment by jacksonw765** [comment]
@aboedo yessir.

**Comment by jacksonw765** [comment]
I started going up the commit history on the develop branch to find where the issue came from. Works fine for me on commit 3fdfa13 however when I try anything past that the issue presents itself. I did a flutter clean and pod update in between each test.

**Comment by aboedo** [comment]
I expect that the linking issues begin in 1.3.0, since that's when we bumped the iOS version to introduce some swift code and modularization - the new project structure is what's causing linking issues with your project. 

what's not obvious to me is why, and why it happens on your project and not the others. 
Have you tried adding a single, empty Swift file? That introduces some extra configuration in the xcode project which might fix it.

**Comment by jacksonw765** [comment]
No I haven't, I can give this a try.

**Comment by jacksonw765** [comment]
That seems to have fixed it! I haven't heard of this trick before but I'll remember this for my future projects. Thanks so much @aboedo

**Comment by aboedo** [comment]
I'm glad it worked! We're still figuring out the kinks of modularizing the framework. @jacksonw765 Thanks for bearing with us!

**Comment by Skyost** [comment]
Hey @aboedo, I still encounter this problem. I did everything you told above (having a Swift file, `pod install --repo-update`, …) but nothing works. I'm on Flutter 1.22.2 and I'm using RevenueCat v1.4.1 (I was previously using v1.2.1 without any problem so far).

<details>
<summary>Podfile</summary>
<pre>
 # Uncomment this line to define a global platform for your project
platform :ios, '9.0'
# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'
project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}
def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end
  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end
require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)
flutter_ios_podfile_setup
target 'Runner' do
  use_frameworks!
  use_modular_headers!
  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
</pre>
</details>

<details>
<summary>Error</summary>
<pre>
    Undefined symbols for architecture armv7:
      "_OBJC_CLASS_$_RCPurchases", referenced from:
          objc-class-ref in PurchasesFlutterPlugin.o
    ld: symbol(s) not found for architecture armv7
    clang: error: linker command failed with exit code 1 (use -v to see
    invocation)
    Undefined symbols for architecture arm64:
      "_OBJC_CLASS_$_RCPurchases", referenced from:
          objc-class-ref in PurchasesFlutterPlugin.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see
    invocation)
</pre>
</details>

**Comment by wupshaw** [comment]
I'm also having this problem @aboedo , I've tried everything mentioned on this thread but still get an error debugging on iOS devices, both real and simulator. Flutter version 1.23.0-18.1.pre with dependency `purchases_flutter: ^1.4.1`

`Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    leveldb-library-flebzexjavoxaweffhxfpgmzvixc
    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_RCPurchases", referenced from:
          objc-class-ref in PurchasesFlutterPlugin.o
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
`

**Comment by aboedo** [comment]
@Skyost @wupshaw this thread is a duplicate of https://github.com/RevenueCat/purchases-flutter/issues/120, let's keep the conversation on that thread. There are also additional steps to try there.
